server/api/team/team.controller.js

```diff
@@ -73,6 +74,66 @@ exports.players = function(req, res) {
     });
 };
 
+exports.fill = function(req, res) {
+    var currentSport = ""
+    var user = req.user;
+      if (user.role !== admin) {
+        logger.log("error", "Filling teams is restricted to admins only");
+        return res.send (403);
+      }
+    Sport.find(req.params.id).then(function(sport) {
+        currentSport = sport
+    }).then(function() {
```

Comment:
Why not just " Sport.find(req.params.id).then(function(currentSport) {" ? Why does currentSport need to be a variable outside the promise chain?